homeassistant/components/zwave/climate.py

```diff
@@ -131,21 +146,46 @@ def update_properties(self):
         if self.values.mode:
             self._hvac_list = []
             self._hvac_mapping = {}
-            hvac_list = self.values.mode.data_items
-            if hvac_list:
-                for mode in hvac_list:
-                    ha_mode = HVAC_STATE_MAPPINGS.get(mode)
+            self._preset_list = []
+            self._preset_mapping = {}
+
+            mode_list = self.values.mode.data_items
+            if mode_list:
+                for mode in mode_list:
+                    ha_mode = HVAC_STATE_MAPPINGS.get(str(mode).lower())
+                    ha_preset = PRESET_MAPPINGS.get(str(mode).lower())
                     if ha_mode and ha_mode not in self._hvac_mapping:
                         self._hvac_mapping[ha_mode] = mode
                         self._hvac_list.append(ha_mode)
-                        continue
-                    self._hvac_list.append(mode)
+                    elif ha_preset and ha_preset not in self._preset_mapping:
+                        self._preset_mapping[ha_preset] = mode
+                        self._preset_list.append(ha_preset)
+                    else:
+                        # If nothing matches
+                        self._preset_list.append(mode)
+
             current_mode = self.values.mode.data
             self._hvac_mode = next(
                 (key for key, value in self._hvac_mapping.items()
-                 if value == current_mode), current_mode)
+                 if value == current_mode), None)
+            self._preset_mode = next(
+                (key for key, value in self._preset_mapping.items()
+                 if value == current_mode), None)
+
+            if self._hvac_mode is not None and self._preset_mode is None:
+                # The current mode is a hvac mode
+                self._preset_mode = PRESET_NONE
+            elif self._hvac_mode is None and self._preset_mode is not None:
+                # The current mode is a preset mode
+                self._hvac_mode = HVAC_MODE_HEAT_COOL
+            elif self._hvac_mode is None and self._preset_mode is None:
+                # The current mode couldn't be found in any mapping
+                self._hvac_mode = HVAC_MODE_HEAT_COOL
+                self._preset_mode = current_mode
```

Comment:
After looking for a more elegant solution, I decided to add this if clause. This solution is the most efficient and at the same time understandable to others.

Reply:
nvm found a better solution